components/script/webdriver_handlers.rs

```diff
@@ -270,6 +272,49 @@ pub fn handle_get_browsing_context_id(
     reply.send(result).unwrap()
 }
 
+fn get_element_in_view_center_point(element: &Element) -> Option<Point2D<i64>> {
+    match element.GetClientRects().iter().next() {
+        Some(rectangle) => {
+            let x = rectangle.X().round() as i64;
+            let y = rectangle.Y().round() as i64;
+            let width = rectangle.Width().round() as i64;
+            let height = rectangle.Height().round() as i64;
+
+            let window = window_from_node(element.upcast::<Node>());
+            let innerWidth = window.InnerWidth() as i64;
+            let innerHeight = window.InnerHeight() as i64;
+
+            let left = cmp::max(0, cmp::min(x, x + width));
+            let right = cmp::min(innerWidth, cmp::max(x, x + width));
+            let top = cmp::max(0, cmp::min(y, y + height));
+            let bottom = cmp::min(innerHeight, cmp::max(y, y + height));
```

Comment:
It's not clear to me that comparing against innerWidth and innerHeight will work when an element is outside the initial viewport and the page is scrolled so that the element is in view.

Reply:
This will not work in that case, it needs to take into account the whole available viewport not just the window. I'll address this issue. However, this case won't come up as scrolling into view is not supported yet.

Reply:
Actually, I think this works. `innerWidth` and `innerHeight` should reflect the size of the whole viewport, not just the size of the window, if I understood correctly.

Reply:
The viewport is the window size.